tests/Controller/ExportControllerTest.php

```diff
@@ -180,4 +181,36 @@ protected function parsePluginVersion(): string
 
         return ltrim($parsed['version'], 'v');
     }
+
+    public function testCorrectTranslatedProductIsExported(): void
+    {
+        $anotherLanguageId = Uuid::randomHex();
+        $this->createLanguage($anotherLanguageId);
+        $this->salesChannelContext = $this->buildSalesChannelContext(
```

Comment:
The sales channel context must be built within the controller. That's what the `SalesChannelContextService` is for, it generates the sales channel context, based on the given shopkey. This means that it must be possible to call the URL https://your-shop.co.uk/findologic?shopkey=shopkey-of-german-channel and the products will be exported in German.

Reply:
@TheKeymaster It is already being built in the `setUp` method, I am only creating a different one here due to a different language test case

Reply:
The sales channel context should not be explicitly set. The controller must be responsible to get it himself based on the shopkey.